File: app/src/main/java/com/hmproductions/bingo/animations/GridCircleAnimation.java
package com.hmproductions.bingo.animations;

import android.view.animation.Animation;
import android.view.animation.Transformation;

import com.hmproductions.bingo.views.CircleView;

public class GridCircleAnimation extends Animation {

    private CircleView circle;

    private float newAngle;

    public GridCircleAnimation(CircleView circle, int newAngle) {
        this.newAngle = newAngle;
        this.circle = circle;
    }

    @Override
    protected void applyTransformation(float interpolatedTime, Transformation transformation) {
        float angle = newAngle * interpolatedTime;
        circle.setAngle(angle);
        circle.requestLayout();
    }
}

File: app/src/main/java/com/hmproductions/bingo/animations/GridCircleAnimation.kt
package com.hmproductions.bingo.animations

import android.view.animation.Animation
import android.view.animation.Transformation
import com.hmproductions.bingo.views.CircleView

class GridCircleAnimation(private val circle: CircleView, newAngle: Int) : Animation() {

    private val newAngle: Float = newAngle.toFloat()

    override fun applyTransformation(interpolatedTime: Float, transformation: Transformation) {
        val angle = newAngle * interpolatedTime
        circle.setAngle(angle)
        circle.requestLayout()
    }
}
